inc/class-sitemaps-index.php

```diff
@@ -0,0 +1,73 @@
+<?php
+/**
+ * Class Core_Sitemaps_Index.
+ * Builds the sitemap index page that lists the links to all of the sitemaps.
+ *
+ * @todo This will probably be split out so that rewrites are in a class, building the xml output is a class,
+ * rendering sitemaps content is a class etc.
+ */
+class Core_Sitemaps_Index {
+
+	/**
+	 * Content of the sitemap to output.
+	 *
+	 * @var string
+	 */
+	protected $sitemap_content = '';
+
+	/**
+	 *
+	 * A helper function to initiate actions, hooks and other features needed.
+	 *
+	 * @uses add_action()
+	 * @uses add_filter()
+	 */
+	public function bootstrap() {
+		add_action( 'init', array( $this, 'url_rewrites' ), 99 );
+		add_filter( 'redirect_canonical', array( $this, 'redirect_canonical' ) );
+		add_filter( 'template_include', array( $this, 'output_sitemap' ) );
+	}
+
+	/**
+	 * Sets up rewrite rule for sitemap_index.
+	 * @todo Additional rewrites will probably need adding to this.
+	 */
+	public function url_rewrites() {
+		add_rewrite_tag( '%sitemap%','sitemap' );
+		add_rewrite_rule( 'sitemap_index\.xml$', 'index.php?sitemap=sitemap', 'top' );
```

Comment:
Let's change this to `sitemap=index` or `sitemap=sitemap_index` so it's more descriptive when we check for this value later in our code.